tests/backup/backup_restore_basic_test.go

```diff
@@ -1168,7 +1172,14 @@ var _ = Describe("{BackupSyncBasicTest}", func() {
 				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of cloud credential named [%s] for org [%s] with [%s] as provider", credName, orgID, provider))
 				log.InfoD("Created Cloud Credentials with name - %s", credName)
 				customBackupLocationName = fmt.Sprintf("autogenerated-backup-location-%v", time.Now().Unix())
-				err = CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, getGlobalBucketName(provider), orgID, "")
+				switch provider {
+				case drivers.ProviderAws:
+					customBucket = fmt.Sprintf("aws-cloudsnapssafewhenbackuplocationdelete-%v", time.Now().Unix())
+					CreateBucket(provider, customBucket)
+				case drivers.ProviderNfs:
+					customBucket = fmt.Sprintf("Cloudsnapssafewhenbackuplocationdeletete-%v", time.Now().Unix())
+				}
+				err = CreateBackupLocation(provider, customBackupLocationName, backupLocationUID, credName, cloudCredUID, customBucket, orgID, "")
```

Comment:
I am wondering do we even need this switch block. We can just pass a string as bucket name and Px-Backup itself will create a bucket for us right?
Is there a need to pre-provision a bucket?
I also made this mistake while fixing `DeleteBackupAndCheckIfBucketIsEmpty`

Let me know what do you think

Reply:
Even for NFS, we are just passing the subpath string and Px-Backup with create the subpath on the fly if it is not there in the NFS backup location

Reply:
In case of object lock this is necessary  hence to make sure it always creates a normal bucket we are doing this.

Reply:
Ok. Can we just convert this piece of code into a function because it can be reused:

```
switch provider {
				case drivers.ProviderAws:
					customBucket = fmt.Sprintf("aws-cloudsnapssafewhenbackuplocationdelete-%v", time.Now().Unix())
					CreateBucket(provider, customBucket)
				case drivers.ProviderNfs:
					customBucket = fmt.Sprintf("Cloudsnapssafewhenbackuplocationdeletete-%v", time.Now().Unix())
				}
```

Reply:
Done